va/va.h

```diff
@@ -2003,6 +2015,142 @@ typedef struct _VAEncMiscParameterParallelRateControl {
 } VAEncMiscParameterParallelRateControl;
 
 /**
+ * \brief MB partition modes and 1/2 1/4 motion search configuration
+ *
+ * Specifies MB partition modes that are disabled. Specifies Half-pel
+ * mode and Quarter-pel mode searching
+ */
+typedef struct _VAEncMiscParameterSubMbPartPel
+{
+    uint32_t disable_inter_sub_mb_partition;
+    union {
+        struct {
+            uint32_t disable_16x16_inter_mb_partition        : 1;
+            uint32_t disable_16x8_inter_mb_partition         : 1;
+            uint32_t disable_8x16_inter_mb_partition         : 1;
+            uint32_t disable_8x8_inter_mb_partition          : 1;
+            uint32_t disable_8x4_inter_mb_partition          : 1;
+            uint32_t disable_4x8_inter_mb_partition          : 1;
+            uint32_t disable_4x4_inter_mb_partition          : 1;
+            uint32_t reserved                                : 1;
+        } bits;
+         uint8_t value;
+    } inter_sub_mb_partition_mask;
+
+    /**
+     * \brief Precison of motion search
+     * 0:Integer mode searching
+     * 1:Half-pel mode searching
+     * 2:Reserved
+     * 3:Quarter-pel mode searching
+     */
+    uint32_t enable_sub_pel_mode;
+    uint8_t sub_pel_mode;
+} VAEncMiscParameterSubMbPartPel;
+
+// per frame encoder controls, once set they will persist for all future frames
+// till it is updated again.
+typedef struct _VAEncMiscParameterEncQuality
+{
+    union
+    {
+        struct
+        {
+            // Use raw frames for reference instead of reconstructed frames.
+            uint32_t useRawPicForRef                    : 1;
```

Comment:
Wondering for what purpose this can be used!
Running only VME in FEI mode is something which can make use of this. But is it useful in general?

Reply:
this structure is for quality tuning without FEI. we can set to useRawPicForRef during ME stage, and also can set FTQ enhanced skip estimation, Downscaling ME ....

Reply:
Currently, middleware implementations only keep references(to make sure it doesn't get destroyed before other frames use it for reference) to the reconstructed surfaces. So,
If the user enables "useRawPicForRef", middleware supposed to keep references to the input raw surface and also to the reconstructed surfaces.  Or does it mean that the middleware is not required to fill the scratch surface in each picture parameter for the reconstructed surface?

Reply:
it depend on driver implementation. if driver keep the map between raw surface & recon surface, when the useRawPicForRef is set, it will find the related input surface by the recon surface index.